ligo/src/token.mligo

```diff
@@ -13,12 +13,29 @@ let call_fa2(param, store : fa2_parameter * storage) : return =
   | Token_metadata_registry (p) -> token_metadata_registry(p, store)
   | Update_operators (p) -> update_operators(p, store)
 
+let authorize_admin (store : storage): storage =
+  if sender = store.admin then
+    store
+  else
+    (failwith("NOT_ADMIN"): storage)
+
 let burn(param, store : burn_param * storage) : return =
-  not_implemented("burn")
+  let valid_storage = authorize_admin(ensure_not_migrated(store)) in
+  let updated_store = debit_from (param.amount, param.from_, param.token_id, store)
+  in (([] : operation list), updated_store)
 
 let mint(param, store : mint_param * storage) : return =
-  not_implemented("mint")
+  let valid_storage = authorize_admin(ensure_not_migrated(store)) in
+  let updated_store = credit_to (param.amount, param.to_, param.token_id, store)
+  in (([] : operation list), updated_store)
 
 let transfer_contract_tokens
     (param, store : transfer_contract_tokens_param * storage) : return =
-  not_implemented("transfer_contract_tokens")
+  match (Tezos.get_entrypoint_opt "%transfer" param.contract_address
```

Comment:
Seems like this is missing an administrator check:
```suggestion
  let valid_storage = authorize_admin(store) in
  match (Tezos.get_entrypoint_opt "%transfer" param.contract_address
```

Reply:
My bad, I totally missed this, I have added the tests to catch this as well.